openapi-validator/src/main/java/org/ballerinalang/openapi/validator/ServiceValidator.java

```diff
@@ -186,18 +299,101 @@ public static void validateResource(OpenAPI openApi,
+                    String failOnErrors = openAPIAnnotation.toString();
+                    if (failOnErrors.trim().equals("false")) {
+                        kind = DiagnosticSeverity.WARNING;
+                        filters.setKind(kind);
+                    }
+                } else if (openAPIAnnotation instanceof ListConstructorExpressionNode) {
+                    ListConstructorExpressionNode list = (ListConstructorExpressionNode) openAPIAnnotation;
+                    String attributeName = specificFieldNode.fieldName().toString().trim();
+                    switch (attributeName) {
+                        case "tags":
+                            List<String> tags = setFilters(list);
+                            filters.setTag(tags);
+                            break;
+                        case "excludeTags":
+                            List<String> eTags = setFilters(list);
+                            filters.setExcludeTag(eTags);
+                            break;
+                        case "operatoins":
+                        case "excludeOperations":
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+        return kind;
+    }
+
+    private static List<String> setFilters(ListConstructorExpressionNode list) {
+        SeparatedNodeList<Node> expressions = list.expressions();
+        Iterator<Node> iterator = expressions.iterator();
+        List<String> tags = new ArrayList<>();
+        while (iterator.hasNext()) {
+            Node item = iterator.next();
+            tags.add(item.toString());
+        }
+        return tags;
+    }
+
     /**
      * Fix the list with openAPIPathSummary by matching resources that documented for validating.
      * @param openAPIPathSummaries      summary list with OpenAPI Path
      * @param resourcePathSummaryList   summary list with resourcePath
      */
     private static void createListOperations(List<OpenAPIPathSummary> openAPIPathSummaries,
-                                             List<ResourcePathSummary> resourcePathSummaryList) {
+                                             Map<String, ResourcePathSummary> resourcePathSummaryList) {
 
-        Iterator<ResourcePathSummary> resourcePathSummaryIterator = resourcePathSummaryList.iterator();
+        Iterator<Map.Entry<String, ResourcePathSummary>> resourcePathSummaryIterator = resourcePathSummaryList
+                .entrySet().iterator();
         while (resourcePathSummaryIterator.hasNext()) {
             boolean isExit = false;
-            ResourcePathSummary resourcePathSummary = resourcePathSummaryIterator.next();
+            ResourcePathSummary resourcePathSummary = resourcePathSummaryIterator.next().getValue();
             for (OpenAPIPathSummary apiPathSummary: openAPIPathSummaries) {
```

Comment:
This for loop doesn't seem like iterating

Reply:
Currently, we hold iterate here due to update the undocumented path in OpenAPI spec. will continue iterate with the next implementation for the undocumented ballerina resource path.